File: Introduction to Data Structures: Hash Table/Practical Application - Hash Map/Isomorphic Strings/IsomorphicStrings.java
class Solution {
    public boolean isIsomorphic(String s, String t) {
        
        HashMap<Character,Character> hashMap = new HashMap<>(); 
        
        for (int i = 0; i < s.length(); i++) {
            if (!hashMap.containsKey(s.charAt(i))) {
                if (!hashMap.containsValue(t.charAt(i))) {
                    hashMap.put(s.charAt(i), t.charAt(i));
                }
                else {
                    return false;
                }
            } else {
                if (hashMap.get(s.charAt(i)) != t.charAt(i)) {
                    return false;
                }
            }
        }
        return true;
    }
}

File: Introduction to Data Structures: Hash Table/3. Practical Application - Hash Map/Isomorphic Strings/IsomorphicStrings.py
class Solution:
    def isIsomorphic(self, s: str, t: str) -> bool:

        hashMap = {}

        for i in range(0, len(s)):

            if s[i] not in hashMap.keys():

                if t[i] not in hashMap.values():

                    hashMap[s[i]] = t[i]
                else:
                    return False

            else:

                val = hashMap[s[i]]

                if val != t[i]:

                    return False

        return True
